spec/03-plugins/26-grpc-web/01-proxy_spec.lua

```diff
@@ -0,0 +1,157 @@
+local cjson = require "cjson"
+local helpers = require "spec.helpers"
+
+-- returns nth byte (0: LSB, 3: MSB if 32-bit)
+local function nbyt(x, n)
+  return bit.band(bit.rshift(x, 8*n), 0xff)
+end
+
+local function be_bytes(x)
+  return nbyt(x, 3), nbyt(x, 2), nbyt(x, 1), nbyt(x, 0)
+end
+
+for _, strategy in helpers.each_strategy() do
+
+  describe("gRPC-Web Proxying [#" .. strategy .. "]", function()
+    local proxy_client
+    local proxy_client_ssl
+
+    local HELLO_REQUEST_TEXT_BODY = "AAAAAAYKBGhleWE="
+    local HELLO_REQUEST_BODY = ngx.decode_base64(HELLO_REQUEST_TEXT_BODY)
+    local HELLO_RESPONSE_TEXT_BODY = "AAAAAAwKCmhlbGxvIGhleWE=" ..
+        "gAAAAB5ncnBjLXN0YXR1czowDQpncnBjLW1lc3NhZ2U6DQo="
+    local HELLO_RESPONSE_BODY = ngx.decode_base64("AAAAAAwKCmhlbGxvIGhleWE=") ..
+        ngx.decode_base64("gAAAAB5ncnBjLXN0YXR1czowDQpncnBjLW1lc3NhZ2U6DQo=")
+
+    lazy_setup(function()
+      local bp = helpers.get_db_utils(strategy, {
+        "routes",
+        "services",
+        "plugins",
+      }, {
+        "grpc-web",
+      })
+
+      local service1 = assert(bp.services:insert {
+        name = "grpc",
+        url = "grpc://localhost:15002",
+      })
+
+      local route1 = assert(bp.routes:insert {
+        protocols = { "http", "https" },
+        paths = { "/" },
+        service = service1,
+      })
+
+      assert(bp.plugins:insert {
+        route = route1,
+        name = "grpc-web",
+        config = {
+          proto = "spec/fixtures/hello.proto",
+        },
+      })
+
+      assert(helpers.start_kong {
+        database = strategy,
+        plugins = "bundled,grpc-web",
+      })
+    end)
+
+    before_each(function()
+      proxy_client = helpers.proxy_client(1000)
+      proxy_client_ssl = helpers.proxy_ssl_client(1000)
+    end)
+
+    lazy_teardown(function()
+      helpers.stop_kong()
+    end)
+
+
+    test("Call gRCP-base64 via HTTP", function()
+      local res, err = proxy_client:post("/hello.HelloService/SayHello", {
+        headers = {
+          ["Content-Type"] = "application/grpc-web-text",
+          ["Content-Length"] = tostring(#HELLO_REQUEST_TEXT_BODY),
+        },
+        body = HELLO_REQUEST_TEXT_BODY,
+      })
+
+      assert.equal(HELLO_RESPONSE_TEXT_BODY, res:read_body())
+      assert.is_nil(err)
+    end)
+
+    test("Call gRCP-bas64 via HTTPS", function()
```

Comment:
```suggestion
    test("Call gRCP-base64 via HTTPS", function()
```

Reply:
for a moment i thought "damn, if it was working with a wrong method name that means the tests are invalid!"...